src/main/java/seedu/address/ui/LessonCard.java

```diff
@@ -44,9 +49,14 @@ public LessonCard(Lesson lesson, int displayedIndex) {
         this.lesson = lesson;
         id.setText(displayedIndex + ". ");
         name.setText(lesson.getName().fullName);
-        deadline.setText("Start time: " + lesson.getTime().value);
-        endTime.setText("End time: " + lesson.getEndTime().value);
-        moduleCode.setText("Module: " + lesson.getModuleCode().moduleCode);
+        DateTimeFormatter inputFormat = DateTimeFormatter.ofPattern(DEADLINE_DATE_TIME_FORMAT)
+                .withResolverStyle(ResolverStyle.STRICT);
+        String date = LocalDateTime.parse(lesson.getTime().value, inputFormat).toLocalDate().format(
+                DateTimeFormatter.ofPattern("dd MMM yyyy"));
+        LocalTime startTime = LocalDateTime.parse(lesson.getTime().value, inputFormat).toLocalTime();
+        LocalTime endTime = LocalDateTime.parse(lesson.getEndTime().value, inputFormat).toLocalTime();
+        time.setText(date + " " + startTime + "-" + endTime);
+        //moduleCode.setText("Module: " + lesson.getModuleCode().moduleCode);
```

Comment:
Good job :) You may want to consider extracting this part into another method as the CS2103T textbook suggests that we should SLAP our code to improve code quality.